pkg/config/data.go

```diff
@@ -111,29 +126,115 @@ var FilePath = func() string {
 // DefaultEndpoint is the default Fastly API endpoint.
 const DefaultEndpoint = "https://api.fastly.com"
 
-// File represents all of the configuration parameters that can end up in the
-// config file. At some point, it may expand to include e.g. user profiles.
+// File represents our dynamic application toml configuration.
 type File struct {
-	Token            string `toml:"token"`
-	Email            string `toml:"email"`
-	Endpoint         string `toml:"endpoint"`
-	LastVersionCheck string `toml:"last_version_check"`
+	Fastly      ConfigFastly              `toml:"fastly"`
+	CLI         ConfigCLI                 `toml:"cli"`
+	User        ConfigUser                `toml:"user"`
+	Language    ConfigLanguage            `toml:"language"`
+	StarterKits ConfigStarterKitLanguages `toml:"starter-kits"`
+}
+
+// ConfigFastly represents fastly specific configuration.
+type ConfigFastly struct {
+	APIEndpoint string `toml:"api_endpoint"`
+}
+
+// ConfigCLI represents CLI specific configuration.
+type ConfigCLI struct {
+	RemoteConfig string `toml:"remote_config"`
+	TTL          string `toml:"ttl"`
+	LastChecked  string `toml:"last_checked"`
+}
+
+// ConfigUser represents user specific configuration.
+type ConfigUser struct {
+	Token string `toml:"token"`
+	Email string `toml:"email"`
+}
+
+// ConfigLanguage represents C@E language specific configuration.
+type ConfigLanguage struct {
+	Rust ConfigRust `toml:"rust"`
+}
+
+// ConfigLanguage represents Rust C@E language specific configuration.
+type ConfigRust struct {
+	// ToolchainVersion is the `rustup` toolchain string for the compiler that we
+	// support
+	ToolchainVersion string `toml:"toolchain_version"`
+
+	// WasmWasiTarget is the Rust compilation target for Wasi capable Wasm.
+	WasmWasiTarget string `toml:"wasm_wasi_target"`
```

Comment:
Does this need to be Dynamic configuration? I don't ever see this changing. cc @acfoltzer

Reply:
It changed in the past from `wasm32-unknown-wasi` to `wasm32-wasi`, and while I doubt it'll change again we might as well be prepared